### PR TITLE
MODSOURMAN-255: Logs show that failed to load files contain 100 records

### PR DESCRIPTION
**Pull request by VRohach**
JIRA-ticket: https://issues.folio.org/browse/MODSOURMAN-255

1. If JobExecution failed - the value of the total records will be 0.
2. The test was updated.

### CODE REVIEW

File: mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java [hunk 1]
```diff
@@ -157,6 +157,7 @@
             jobExecution.setUiStatus(JobExecution.UiStatus.fromValue(Status.valueOf(status.getStatus().name()).getUiStatus()));
             if (status.getStatus().equals(StatusDto.Status.ERROR)) {
               jobExecution.withErrorStatus(JobExecution.ErrorStatus.fromValue(status.getErrorStatus().name()))
+                .withProgress(jobExecution.getProgress().withTotal(0))
```
Comment (KaterynaSenchenko) [inline mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java]: Please update progress in place where errorStatus is set for JobExecution (check if it is FILE_PROCESSING_ERROR).

File: mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java [hunk 2]
```diff
@@ -443,4 +439,19 @@ private Future saveSnapshots(List<Snapshot> snapshots, OkapiConnectionParams par
...
+   * Updates jobExecution object, if Error exists.
+   * @param status - DTO which contains new status
+   * @param jobExecution - specific JobExecution
+   */
+  private void updateJobExecutionIfErrorExist(StatusDto status, JobExecution jobExecution) {
+    if (status.getStatus().equals(StatusDto.Status.ERROR)) {
+      jobExecution.withErrorStatus(JobExecution.ErrorStatus.fromValue(status.getErrorStatus().name()))
+        .withCompletedDate(new Date());
+      if(jobExecution.getErrorStatus().equals(JobExecution.ErrorStatus.FILE_PROCESSING_ERROR)){
+        jobExecution.withProgress(jobExecution.getProgress().withTotal(0));
```
Comment (KaterynaSenchenko) [inline mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java]: pls use setter here as well

File: mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java [hunk 3]
```diff
@@ -443,4 +439,19 @@ private Future saveSnapshots(List<Snapshot> snapshots, OkapiConnectionParams par
...
     return future;
   }
+
+  /**
+   * Updates jobExecution object, if Error exists.
+   * @param status - DTO which contains new status
+   * @param jobExecution - specific JobExecution
+   */
+  private void updateJobExecutionIfErrorExist(StatusDto status, JobExecution jobExecution) {
+    if (status.getStatus().equals(StatusDto.Status.ERROR)) {
```
Comment (KaterynaSenchenko) [inline mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java]: just a note, we can use '==' with enums

File: mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java [hunk 4]
```diff
@@ -443,4 +439,19 @@ private Future saveSnapshots(List<Snapshot> snapshots, OkapiConnectionParams par
...
+      jobExecution.withErrorStatus(JobExecution.ErrorStatus.fromValue(status.getErrorStatus().name()))
```
[the remark this answers is not in the document]
Comment (VRohach) [reply mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java]: Sure, I will do it. It was just copied from the previous place where this logic situated before extracting to this method

File: mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java (hunk 3 above)
Comment (VRohach) [reply mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java]: Done

File: mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java (hunk 2 above)
Comment (VRohach) [reply mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java]: Done

**Comment by VRohach** [comment]
> Could you also add a test for this check? Thanks

Done. Could you please review it ?

**Pull request merged**